pandas/core/internals/blocks.py

```diff
@@ -2250,6 +2244,8 @@ def __init__(self, values, placement, ndim=2, dtype=None):
         # and just use DatetimeBlock's.
         if dtype is not None:
             values = self._maybe_coerce_values(values, dtype=dtype)
+            # TODO: this gets hit in msgpack tests, but in all cases the values
+            #  are already DatetimeArrays with the appropriate dtype
```

Comment:
this comment should be removed before merging, but if someone well-informed says its OK we may be able to get rid of this check and get some further simplification.

Reply:
where does this get hit?

Reply:
in the tests the only place this comes up is in test_packers

Reply:
you can certainly do this at a higher level. in packers you know the block type & have access to the ``_holder``; can just construct it and pass fully formed values, then don't need to pass the dtype thru

Reply:
Yah thats what Im thinking now, and it may actually make some of the other 24024 follow-ups easier.  Anyway, just removed the comment.